src/v/cluster/rm_stm.cc

```diff
@@ -2645,6 +2670,44 @@ std::ostream& operator<<(std::ostream& o, const rm_stm::abort_snapshot& as) {
     return o;
 }
 
+void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
+    switch (type) {
+    case rm_stm::clear_type::tx_pids: {
+        if (
+          _log_state.fence_pid_epoch.size() <= _max_concurrent_producer_ids()) {
+            return;
+        }
+        break;
+    }
+    case rm_stm::clear_type::idempotent_pids: {
+        if (
+          _log_state.lru_idempotent_pids.size()
+          <= _max_concurrent_producer_ids()) {
+            return;
+        }
+        break;
+    }
+    }
+
+    ssx::spawn_with_gate(_gate, [this, type] {
+        return _state_lock.hold_read_lock().then(
+          [this,
+           type](ss::basic_rwlock<>::holder unit) mutable -> ss::future<> {
+              switch (type) {
+              case rm_stm::clear_type::tx_pids: {
+                  vlog(_ctx_log.trace, "Gonna clear old tx pids");
+                  return clear_old_tx_pids().finally([u = std::move(unit)] {});
+              }
+              case rm_stm::clear_type::idempotent_pids: {
+                  vlog(_ctx_log.trace, "Gonna clear old idempotent pids");
```

Comment:
maybe clean up the wording on these log statements to be more formal and informational?